cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -125,13 +126,13 @@ public static void validateConfig(SecureStore secureStore, SourceControlConfig s
           .setTags(false);
       validateDefaultBranch(cmd.callAsMap(), config.getDefaultBranch());
     } catch (TransportException e) {
-      throw new RepositoryConfigValidationException("Failed to connect with remote repository: " + e.getMessage(), e);
+      throw new RemoteRepositoryValidationException("Failed to connect with remote repository: " + e.getMessage(), e);
```

Comment:
Even authentication failures will result in a TransportException.

Reply:
Return 400 now